sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/TableFileCatalog.scala

```diff
@@ -42,24 +43,21 @@ class TableFileCatalog(
 
   protected val hadoopConf = sparkSession.sessionState.newHadoopConf
 
+  private val fileStatusCache = FileStatusCache.getOrInitializeShared(new Object(), sparkSession)
```

Comment:
Does the file status cache need to be private to an instance of `TableFileCatalog`? Can we use the table's qualified name as the key here instead?